### PR TITLE
Remove redundant ShowAllCommand

### PR DESCRIPTION
**Pull request by hjungwoo01**
Previously, `ShowAllCommand` and `ListCommand`
duplicated functionality, leading to confusion.
To improve user experience, `ShowAllCommand` was
removed, ensuring `ListCommand` suffices for
listing all entities.

Changes made:
- Removed `ShowAllCommand` to prevent function overlap with `ListCommand`.
- Tested `ListCommand` to confirm it meets all listing needs.

This simplification enhances usability, making the application more intuitive and efficient. Now,
`ListCommand` alone offers a complete listing
solution, streamlining the command suite.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/ListCommandTest.java [hunk 1]
```diff
@@ -36,4 +41,13 @@ public void execute_listIsFiltered_showsEverything() {
         showEmployeeAtIndex(model, INDEX_FIRST_EMPLOYEE);
         assertCommandSuccess(new ListCommand(), model, ListCommand.MESSAGE_SUCCESS, expectedModel);
     }
+
```
Comment (shayaansultan) [inline src/test/java/seedu/address/logic/commands/ListCommandTest.java:44]: Good code coverage

**Review by shayaansultan — COMMENTED** [review]
LGTM!

File: src/test/java/seedu/address/logic/commands/ListCommandTest.java [hunk 2]
```diff
@@ -36,4 +41,13 @@ public void execute_listIsFiltered_showsEverything() {
...
+    @Test
+    public void execute_list_success() {
+        ModelStub modelStub = new ModelStubWithFilteredEmployeeList();
+        ListCommand listCommand = new ListCommand();
+
+        assertDoesNotThrow(() -> listCommand.execute(modelStub));
+        assertEquals(getTypicalEmployees(), modelStub.getFilteredEmployeeList());
+    }
```
Comment (PyromancerBoom) [inline src/test/java/seedu/address/logic/commands/ListCommandTest.java:52]: Nice!

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -85,9 +84,6 @@ public Command parseCommand(String userInput) throws ParseException {
         case FilterCommand.COMMAND_WORD:
             return new FilterCommandParser().parse(arguments);
 
-        case ShowAllCommand.COMMAND_WORD:
-            return new ShowAllCommand();
-
         case AddTaskCommand.COMMAND_WORD:
             return new AddTaskCommandParser().parse(arguments);
```
Comment (PyromancerBoom) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:88]: I like the abstraction of the command word itself!

File: src/test/java/seedu/address/model/employee/NameTest.java [hunk 4]
```diff
@@ -57,4 +59,30 @@ public void equals() {
...
+    }
+
+    @Test
+    public void hashCode_differentValue_hashCodeNotEqual() {
+        Name name1 = new Name("Jon");
+        Name name2 = new Name("Doe");
+
+        assertNotEquals(name1.hashCode(), name2.hashCode(),
+                "Hash code should be different for objects with different values");
+    }
```
Comment (PyromancerBoom) [inline src/test/java/seedu/address/model/employee/NameTest.java:87]: Quite thorough testcases!

File: src/test/java/seedu/address/model/employee/UniqueEmployeeListTest.java [hunk 5]
```diff
@@ -173,4 +174,53 @@ public void asUnmodifiableObservableList_modifyList_throwsUnsupportedOperationEx
...
+
+    @Test
+    public void setEmployees_listWithExistingAndNewEmployees_success() {
+        uniqueEmployeeList.add(ALICE);
+        List<Employee> newEmployees = Arrays.asList(ALICE, BOB);
+        uniqueEmployeeList.setEmployees(newEmployees);
+        UniqueEmployeeList expectedUniqueEmployeeList = new UniqueEmployeeList();
+        expectedUniqueEmployeeList.add(ALICE);
+        expectedUniqueEmployeeList.add(BOB);
+        assertEquals(expectedUniqueEmployeeList, uniqueEmployeeList);
```
Comment (PyromancerBoom) [inline src/test/java/seedu/address/model/employee/UniqueEmployeeListTest.java:211]: Pretty good coverage! Maybe some spacing could improve readability!

**Review by PyromancerBoom — COMMENTED** [review]
LGTM!

**Pull request merged**